src/components/Common/FolderContextMenu.tsx

```diff
@@ -1,69 +1,91 @@
-import { IconPencilMinus, IconTrashX } from '@tabler/icons-react';
+import {
+  IconDots,
+  IconFolderPlus,
+  IconPencilMinus,
+  IconTrashX,
+} from '@tabler/icons-react';
 import { MouseEventHandler } from 'react';
 
 import { useTranslation } from 'next-i18next';
 
+import classNames from 'classnames';
+
 import { HighlightColor } from '@/src/types/components';
 
-import DotsIcon from '../../../public/images/icons/dots-vertical.svg';
 import { Menu, MenuItem } from './DropdownMenu';
 
 interface FolderContextMenuProps {
-  onDelete: MouseEventHandler<unknown>;
-  onRename: MouseEventHandler<unknown>;
+  onDelete?: MouseEventHandler<unknown>;
+  onRename?: MouseEventHandler<unknown>;
+  onAddFolder?: MouseEventHandler;
   highlightColor: HighlightColor;
 }
 export const FolderContextMenu = ({
   onDelete,
   onRename,
+  onAddFolder,
   highlightColor,
 }: FolderContextMenuProps) => {
   const { t } = useTranslation('sidebar');
 
+  const highlightBg = classNames(
+    highlightColor === 'green' ? 'hover:bg-green/15' : 'hover:bg-violet/15',
+  );
+
+  if (!onDelete && !onRename && !onAddFolder) {
+    return null;
+  }
+
   return (
     <Menu
       type="contextMenu"
       className="justify-self-end"
       trigger={
-        <DotsIcon
-          className="rotate-90 text-gray-500"
-          width={18}
-          height={18}
+        <IconDots
+          className="text-gray-500"
           size={18}
```

Comment:
if we don't want to see difference with previous icon you need to decrease size
```suggestion
          size={16}
```